base/dates/parse.jl

```diff
@@ -281,9 +281,14 @@ function Base.tryparse{T<:TimeType}(
     pos, len = start(str), endof(str)
     values, pos = tryparsenext_internal(T, str, pos, len, df, false)
     if isnull(values)
-        Nullable{T}()
+        return Nullable{T}()
     else
-        Nullable{T}(T(unsafe_get(values)...))
+        try
+            parsed = Nullable{T}(T(unsafe_get(values)...))
```

Comment:
Makes sense. We could also pass `Exception` or a similar type as the first argument to `isvalid` to enable this behavior (with the idea that the non-throwing behavior would be equivalent to passing `Bool`).